src/systems/buoyancy/Buoyancy.cc

```diff
@@ -477,6 +472,13 @@ void Buoyancy::PreUpdate(const ignition::gazebo::UpdateInfo &_info,
       else if (this->dataPtr->buoyancyType
         == BuoyancyPrivate::BuoyancyType::GRADED_BUOYANCY)
       {
+        if (newPose)
+        {
+          // Skip entity if WorldPose and inertial are not yet ready
+          // TODO(arjo): Find a way of disabling gravity effects for
+          // this first iteration.
```

Comment:
I think we may be able to avoid this skip if we use `gazebo::worldPose` instead of the `WorldPose` component. That function should provide the world pose straight away.

Reply:
Hmm, but we need to use `link::WorldInertialPose`...

Reply:
Ahh I see, maybe the implementation of that method can be improved :thinking: